ops/services/app_service/metabase/main.tf

```diff
@@ -1,3 +1,75 @@
+    '${data.azurerm_key_vault_secret.postgres_nophi_pass.value}';
+    GRANT USAGE ON SCHEMA ${local.schema_name} TO ${data.azurerm_key_vault_secret.postgres_nophi_user.value};
+    GRANT SELECT ON TABLE facility_device_type TO ${data.azurerm_key_vault_secret.postgres_nophi_user.value};
+    GRANT SELECT ON TABLE device_type TO ${data.azurerm_key_vault_secret.postgres_nophi_user.value};
+    GRANT SELECT ON TABLE organization TO ${data.azurerm_key_vault_secret.postgres_nophi_user.value};
+    GRANT SELECT ON TABLE patient_answers TO ${data.azurerm_key_vault_secret.postgres_nophi_user.value};
+    GRANT SELECT ON TABLE facility TO ${data.azurerm_key_vault_secret.postgres_nophi_user.value};
+    GRANT SELECT ON TABLE patient_link TO ${data.azurerm_key_vault_secret.postgres_nophi_user.value};
+    GRANT SELECT ON TABLE data_hub_upload TO ${data.azurerm_key_vault_secret.postgres_nophi_user.value};
+    GRANT SELECT ON TABLE time_of_consent TO ${data.azurerm_key_vault_secret.postgres_nophi_user.value};
+    GRANT SELECT ON TABLE specimen_type TO ${data.azurerm_key_vault_secret.postgres_nophi_user.value};
+    GRANT SELECT ON TABLE device_specimen_type TO ${data.azurerm_key_vault_secret.postgres_nophi_user.value};
+    GRANT SELECT ON TABLE facility_device_specimen_type TO ${data.azurerm_key_vault_secret.postgres_nophi_user.value};
+    GRANT SELECT ON TABLE test_order TO ${data.azurerm_key_vault_secret.postgres_nophi_user.value};
+
+    GRANT SELECT (login_email, internal_id, created_at, updated_at, last_seen, is_deleted) ON TABLE api_user
+    TO ${data.azurerm_key_vault_secret.postgres_nophi_user.value};
+    
+    GRANT SELECT (internal_id, created_at, created_by, updated_at, updated_by, is_deleted, provider_id)
+    ON TABLE provider TO ${data.azurerm_key_vault_secret.postgres_nophi_user.value};
+                
+    GRANT SELECT (internal_id, created_at, created_by, updated_at, updated_by, is_deleted, organization_id, employed_in_healthcare,
+    resident_congregate_setting, facility_id, race, gender, ethnicity, lookup_id, role)
+    ON TABLE person TO ${data.azurerm_key_vault_secret.postgres_nophi_user.value};
+                
+    GRANT SELECT (device_specimen_type_id, created_at, created_by, updated_at, updated_by, patient_id, organization_id, 
+    device_type_id, result, facility_id, survey_data, date_tested_backdate, test_order_id, correction_status, 
+    prior_corrected_test_event_id, internal_id, reason_for_correction)
+    ON TABLE test_event TO ${data.azurerm_key_vault_secret.postgres_nophi_user.value};
+  "
+  EOF
+}
+
+variable "app_settings_overrides" {
+  description = "override values for app_settings"
+  default     = {}
+}
+
+resource "null_resource" "add_nophi_db_user" {
+  triggers = {
+    always_run = timestamp()
+  }
+
+  provisioner "local-exec" {
+    command    = var.run_nophi_user_create == false ? "echo" : local.grant_command
+    on_failure = continue
```

Comment:
We kind of glossed over this on the copy/paste but is this true? Should we continue deployment if this command fails?

Reply:
Probably not, because if it fails the rest of the metabase setup will theoretically also fail. I changed this to a `fail` but something to keep in mind is that I think this should run regardless of which metabase version is being created (regular or readonly). We probably don't want it to fail if it's creating the regular version of metabase, but I'm not sure if there's a way to implement that conditional logic here.

Reply:
Yeah I agree, this should run regardless to be as uniform across environments as possible. I think there's also an argument to be made that if we can't proceed on failure for the read-only Metabase, then it's important to test that on every deploy - otherwise we could break it and not find about it until later or happened to see it in the logs, since we'd be ignoring the failure.